packages/core/src/UserState.ts

```diff
@@ -157,7 +157,25 @@ export default class UserState extends Synchronizer {
         return foundLeaf.index
     }
 
-    async getEpochKeys(epoch: bigint) {
+    getEpochKeys(_epoch?: bigint | number, nonce?: number) {
+        const epoch = _epoch ?? this.calcCurrentEpoch()
+        if (
+            typeof nonce === 'number' &&
+            nonce >= this.settings.numEpochKeyNoncePerEpoch
+        ) {
+            throw new Error(
+                `getEpochKeys nonce ${nonce} exceeds max nonce value ${this.settings.numEpochKeyNoncePerEpoch}`
```

Comment:
I think we can throw error with prefix like `Synchronizer`
https://github.com/Unirep/Unirep/blob/36a62ffb788bcc109b48ab598ea3f5f6a90066b9/packages/core/src/Synchronizer.ts#L41
But it can be done in another PR since there are many errors without prefix